tracing-forest/tests/simulate_connections.rs

```diff
@@ -0,0 +1,93 @@
+//! This test simulates processing many clients on multithreaded task, where
+//! each client has many `await` points.
+//!
+//! It is intended to maximize the amount of concurrent operations to demonstrate
+//! that `tracing-forest` does, in fact, keep each one coherent.
+use rand::Rng;
+use tokio::time::{sleep, Duration};
+use tracing::{info, trace_span, Instrument};
+
+type Result<T> = core::result::Result<T, Box<dyn std::error::Error>>;
+
+async fn sleep_rand() {
+    let millis = rand::thread_rng().gen_range(10..200);
+    sleep(Duration::from_millis(millis)).await
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_n_tasks_random_sleeps() -> Result<()> {
+    let num_clients = 10;
+
+    let logs = tracing_forest::capture()
+        .set_global(true)
+        .build()
+        .on(async {
+            let mut clients = Vec::with_capacity(10);
+            for client in 0..num_clients {
+                let handle = tokio::spawn(
```

Comment:
Does it also work with spawn_blocking?

Reply:
The subscriber is being set globally, so it should work no matter what. Is there anything that `spawn_blocking` does that might make you think otherwise?

Reply:
Spawn blocking creates new threads, rather than using the normal tokio thread pool for tasks :)

Reply:
I used `#[tokio::main(flavor = "multi_thread")]` under the assumption that it was creating new threads, so wouldn't this case already be handled? I can also just add another integration test if not.

Reply:
multi-thread means tasks can run from a thread pool which IS a good test. but spawn-blocking makes new threads outside of that thread pools.